_posts/2020-03-24-dask-developer-workshop.markdown

```diff
@@ -0,0 +1,90 @@
+
+[*] quote Matthew Rocklin
+
+## The Blue Yonder way of using Dask
+
+We at Blue Yonder are still finishing our migration from a proprietary job scheduler to Dask.distributed. This is why when _we_ talk about _Dask_ in this post, we are mainly referring to _Dask.distributed_, as this is our main use case.
+
+We quickly realized at the workshop that Blue Yonder’s use case is almost unique in the Dask community.
+
+Florian Jetter opened the stage at the conference and presented the typical data flow for our machine learning products, the usage of Dask and Dask.distributed within this flow, and where we were currently facing issues.
+ 
+At Blue Yonder, we provide Machine Learning as a Service to our customers in retail. The data which our customers provide through an API is inserted into a relational database. The Machine Learning and prediction steps need a denormalized data format, which we provide in form of Parquet datasets in the Azure Blob Store. The resulting predictions are written back to Parquet files and offered to the customer through an API. 
+The Data Engineering team leverages the distributed scheduler for parallel task execution during the data transformation steps between database and Parquet datasets. Most parts of these pipelines use Dask bags or delayed objects for map/reduce operations. Dask dataframes are especially useful when we reshuffle datasets with an existing partitioning to a differently partitioned dataset. We use [Kartothek](https://github.com/JDASoftwareGroup/kartothek) for this purpose.
+
+A lot of Dask users we met at the workshop also utilized Dask clusters for data-heavy calculations. However, the Blue Yonder use case remained special: most other users were working in an R&D-like environment. Their computations are not running in comparable production environments. Most importantly, these users had no service level agreements with customers about, for example, delivery times for predictions. It seems that Blue Yonder puts higher requirements on the stability and robustness of the distributed scheduler than any other of the users present at the conference.
+
+## Issues we have encountered with Dask and potential improvements
+
+In these discussions we talked about a lot of topics, some of which have been or still are big problems for us. We encountered community members who had similar experiences. Below are some of the "rants" we want to share.
+
+
+### Distributed stability
+We migrated some of our largest data pipelines to Dask in the last months of 2019. For these pipelines, we started to observe significant instability during the computation of the Dask graph. We hadn't seen this issue in any of our previous pipelines running on distributed. Our team contributed several patches upstream in order to resolve these issues on our side. 
```

Comment:
```suggestion
We migrated some of our largest data pipelines to Dask in the last months of 2019. For these pipelines, we started to observe significant instability during the computation of the Dask graph. We hadn't seen this issue in any of our previous pipelines running on distributed. To resolve these issues on our side, our team ended up contributing several patches upstream. 
```

Reply:
"physicists English" according to @nefta-kanilmaz-jdas, I don't have strong feelings about it tho

Reply:
:P Not every reader is necessarily going to be a physicist, though, thus this suggestion.